utils/filter.c

```diff
@@ -113,8 +113,15 @@ struct uftrace_filter *uftrace_match_filter(uint64_t ip, struct rb_root *root,
 		iter = rb_entry(parent, struct uftrace_filter, node);
 
 		if (match_ip(iter, ip)) {
+#if defined(__i386__)
+			__asm__ __volatile__ ("sub $8, %esp\n");
+			__asm__ __volatile__ ("movsd %xmm0, (%esp)\n");
+#endif
 			memcpy(tr, &iter->trigger, sizeof(*tr));
```

Comment:
this is a problem as memcpy() can corrupt SSE registers as you saw.  actually libmcount uses its own memcpy implementation but this code is used both for libmcount and uftrace.

So I suggest copying by assignment instead of memcpy since we build libmcount with -mno-sse2.

```
*tr = iter->trigger;
```

Reply:
got it!